src/auto-evo/simulation/foodsource/PhotosyntheticFoodSource.cs

```diff
@@ -0,0 +1,51 @@
+﻿public class PhotosyntheticFoodSource : IFoodSource
+{
+    private readonly Compound sunlight = SimulationParameters.Instance.GetCompound("sunlight");
+    private readonly Compound glucose = SimulationParameters.Instance.GetCompound("glucose");
+    private readonly Compound atp = SimulationParameters.Instance.GetCompound("atp");
+
+    private float totalSunlight;
+
+    public PhotosyntheticFoodSource(Patch patch)
+    {
+        totalSunlight = patch.Biome.Compounds[sunlight].Dissolved * Constants.AUTO_EVO_SUNLIGHT_ENERGY_AMOUNT;
+    }
+
+    public float FitnessScore(Species species)
+    {
+        var microbeSpecies = (MicrobeSpecies)species;
+
+        var photosynthesisingScore = 0.0f;
+        foreach (var organelle in microbeSpecies.Organelles)
+        {
+            foreach (var process in organelle.Definition.RunnableProcesses)
+            {
+                if (process.Process.Inputs.ContainsKey(sunlight))
+                {
+                    if (process.Process.Outputs.ContainsKey(glucose))
+                    {
+                        photosynthesisingScore += process.Process.Outputs[glucose]
+                            / process.Process.Inputs[sunlight] / Constants.AUTO_EVO_GLUCOSE_USE_SCORE_DIVISOR;
+                    }
+
+                    if (process.Process.Outputs.ContainsKey(atp))
+                    {
+                        photosynthesisingScore += process.Process.Outputs[atp]
+                            / process.Process.Inputs[sunlight] / Constants.AUTO_EVO_ATP_USE_SCORE_DIVISOR;
+                    }
```

Comment:
I don't remember any process producing directly atp from light, so this case is no useful yet, is it (seems like a copy-paste of chemosynthetic)?

Reply:
Normally I would be in the "don't futureproof" camp and want to delete this, but since a sunlight to ATP organelle _could_ exist just by changing a JSON file, I think this currently useless code is justified.

Reply:
I'd think so at first, but this organelle might use sunlight + other compound, which you do not account for. So I'd argue in favor of either making it stronger or scraping it all the like.

Reply:
You drive a hard bargain. Consider it gone.

Reply:
Besides, I think it's technically not photosynthesis by definition if it doesn't make Glucose.

Reply:
What if all of these were generalized so that the input compounds for all processes making ATP were taken into account, instead of keeping these hardcoded categories?

Reply:
The danger of that is losing control of weights for other factors. We had already discussed earlier in this PR things like rust-eaters that sporadically move, and I would like to give the "less active plants are better" rule that I just took out. At this point, the photosynthesis are coming closer and closer together, so maybe they should merge, but I have a hunch in the longer term we will want to nudge the two to evolve differently in other ways.

Reply:
Well, then what about having the general step before that with manually coded tweak values for the specific compound types? (These could perhaps be defined in compounds.json for easy experimenting)